apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SessionPoolManager.cs

```diff
@@ -161,12 +162,20 @@ internal void Release(SessionPool sessionPool)
         internal IReadOnlyList<Statistics> GetStatistics() =>
             _targetedPools.ToArray().Select(tp => tp.Value.GetStatisticsSnapshot()).ToList().AsReadOnly();
 
-        internal SessionPool.DatabaseStatistics GetDatabaseStatistics(SpannerClientCreationOptions options, DatabaseName databaseName)
+        internal SessionPoolKeyStatistics GetDatabaseStatistics(SpannerClientCreationOptions options, SessionPoolKey key)
         {
             GaxPreconditions.CheckNotNull(options, nameof(options));
-            GaxPreconditions.CheckNotNull(databaseName, nameof(databaseName));
             _targetedPools.TryGetValue(options, out var targetedPool);
-            return targetedPool?.SessionPoolOrNull?.GetStatisticsSnapshot(databaseName);
+            return targetedPool?.SessionPoolOrNull?.GetStatisticsSnapshot(GaxPreconditions.CheckNotNull(key, nameof(key)));
+        }
+
+        [Obsolete("", error: false)]
+        internal SessionPool.DatabaseStatistics GetDatabaseStatistics(SpannerClientCreationOptions options, DatabaseName databaseName)

```

Comment:
This method is internal, it can be removed and callers can now call the new one.

Reply:
This cannot be removed for now, as it is being used in SpannerConnection.GetSessionPoolDatabaseStatistics(). If we remove  this then there is no way to get DatabaseStatistics from SessionPoolManger. Although we can get SessionPoolKeyStatistics from its new overload, but then will not be able to converted it to DatabaseStatistics. Atleast not through DatabaseStatistics  contructor as its internal. Let me know your thoughts on the same.

Reply:
Ah yes, you are right. I had forgotten that the Statistics classes are in the other package. But then please, even though it's internal, leave a reason for it being obsolete. Something like, "we still need to call this method for implementing the obsolete SpannerConnection.GetSessionPoolDatabaseStatistics()".